app/src/main/java/com/quran/labs/androidquran/data/QuranInfo.java

```diff
@@ -259,6 +259,11 @@ private String getSuraNameFromPage(Context context, int page) {
   }
 
   public int getJuzFromPage(int page) {
+    if (page == 121) {
+      return 6;
+    } else if (page == 201) {
+      return 10;
+    }
```

Comment:
sorry, if it is required to be changed, why not change the data source directly (eg. in this file https://github.com/quran/quran_android/blob/master/common/data/src/main/java/com/quran/data/pageinfo/common/MadaniDataSource.kt) 🤔

Reply:
Because the data is not wrong. Juz' 7, for example, indeed starts at page 121 but on the last aya in the page. The only thing that should be changed is when the user will be notified about the end of the Juz. In this example, Juz' 6 does not end in page 121 but in 122. If you write Juz' 7 on page 121, the user might think that Juz' 7 starts on the beginning of this page which is not the case.
Another scenario to be considered is if one would like to jump to the 7th Juz. In this case the app should go to page 121 and not 122 because it starts at the end of page 121.

Reply:
Maybe a cleaner way of doing it would be having another array getPageForJuzArray2? with the two pages changed and adding a flag to the function to specify whether it should be the original one or the second one but I thought that this is too much for it?

Reply:
personally I prefer there is another array.
I think current code has many hidden information that can't be easily understood just by reading the code